source/geometries/GenericWLSFiber.h

```diff
@@ -0,0 +1,87 @@
+// -----------------------------------------------------------------------------
+//  nexus | GenericWLSFiber.h
+//
+//  Geometry of a configurable wave-length shifting optical fiber.
+//
+//  The NEXT Collaboration
+// -----------------------------------------------------------------------------
+
+#ifndef GENERIC_WLS_FIBER_H
+#define GENERIC_WLS_FIBER_H
+
+#include "GeometryBase.h"
+#include <G4MaterialPropertyVector.hh>
+
+class G4Material;
+class G4GenericMessenger;
+class G4MaterialPropertiesTable;
+
+namespace nexus {
+
+  class GenericWLSFiber: public GeometryBase
+  {
+
+  public:
+    // Constructor for a generic wave-length shifting optical fiber
+    GenericWLSFiber(G4String    name,
+                    G4String    shape,            // "round"  or "square"
+                    G4double    thickness,        // diameter or side
+                    G4double    length,
+                    G4int       num_claddings,
```

Comment:
Considering that, in practice, there are only two options (single or double cladding), I'd make this either an _enum_ or a _boolean_.